app/models/creatorship.rb

```diff
@@ -44,7 +44,7 @@ def check_invalid
   # Make sure that if this is an invitation, we're not inviting someone who has
   # disabled invitations.
   def check_disallowed
-    return if approved? || pseud.nil?
+    return if approved? || pseud.nil? || pseud&.user&.banned || pseud&.user&.suspended
```

Comment:
It took me a sec to figure out what this was fixing. Can we put the `pseud&.user&.banned || pseud&.user&.suspended` check on its own line with a comment about what it does, e.g.,

```
# Prevent this error from appearing in addition to the error for banned or suspended users.
return if pseud&.user&.banned || pseud&.user&.suspended
```

Reply:
This approach requires repeating the banned/suspended check in two different functions, but it might be nice to take a different approach that avoids that repetition.

I'd be inclined to change the order of the validations so that `check_banned` runs before `check_disallowed`, and either toss in a call to `throw :abort` at the end of `check_banned` to stop any further error messages from being generated for the `Creatorship`, or modify the `check_disallowed` check so that it only runs if there are no other errors.

Reply:
> It took me a sec to figure out what this was fixing. Can we put the `pseud&.user&.banned || pseud&.user&.suspended` check on its own line with a comment about what it does, e.g.,
> 
> ```
> # Prevent this error from appearing in addition to the error for banned or suspended users.
> return if pseud&.user&.banned || pseud&.user&.suspended
> ```

Changes made and submitted. I am willing to consider ticklinginstant's suggestion but my approach had been to create a simplistic solution in the short term and not try to re-write the entire block. up to either.

Reply:
If you're up for it, I think ticking's suggestion of reordering the validations adding `throw :abort` is a good one. We're still working on a few issues for our current release, so there's no rush -- it'll probably be at least 5-7 days before we start merging issues for the next release.

Reply:
Sorry for the delay. I'll take a look at it and see what I can do!